### PR TITLE
[Core] Expose `HasMasterSlaveConstraint` from `ModelPart` to python

### PR DESCRIPTION
**Pull request by loumalouomega**
**📝 Description**

Expose `HasMasterSlaveConstraint` from `ModelPart` to python +  adding test

**🆕 Changelog**

- [Expose HasMasterSlaveConstraint from ModelPart to python](https://github.com/KratosMultiphysics/Kratos/commit/063a668025c7d95ccdf13c3bb67a0dfa030ab201)
- [Adding test + doc](https://github.com/KratosMultiphysics/Kratos/commit/1d8c57d596d4f668ba31018853cb366d0943f738)

### CODE REVIEW

File: kratos/python/add_model_part_to_python.cpp [hunk 1]
```diff
@@ -938,6 +935,9 @@ void AddModelPartToPython(pybind11::module& m)
         .def("GetNonHistoricalVariablesNames", [](ModelPart& rModelPart, ModelPart::ConditionsContainerType& rContainer, bool doFullSearch=false) -> std::unordered_set<std::string> {
             return GetNonHistoricalVariablesNames(rModelPart, rContainer, doFullSearch);
         })
+        .def("HasMasterSlaveConstraint", [](ModelPart& rModelPart, ModelPart::IndexType MasterSlaveConstraintId) -> bool {
+            return rModelPart.HasMasterSlaveConstraint(MasterSlaveConstraintId);
+        })
```
Comment (rubenzorrilla) [inline kratos/python/add_model_part_to_python.cpp:940]: Can't we do it without the lambda in this case?

Comment (loumalouomega) [reply kratos/python/add_model_part_to_python.cpp:940]: No, because of the mesh index...

Comment (rubenzorrilla) [reply kratos/python/add_model_part_to_python.cpp:940]: Argggg... :rage:

**Review by rubenzorrilla — APPROVED** [review]

File: kratos/python/add_model_part_to_python.cpp (hunk 1 above)
Comment (loumalouomega) [reply kratos/python/add_model_part_to_python.cpp:940]: Mesh index delenda est

**Pull request merged**